### PR TITLE
Add configuration option to override/transform the scraped mBean value

### PR DESCRIPTION
**Pull request by icetrain**

### CODE REVIEW

**Comment by icetrain** [comment]
Solves #47

File: collector/src/main/java/io/prometheus/jmx/JmxCollector.java [hunk 1]
```diff
@@ -240,38 +245,70 @@ public void recordBean(
...
               continue;
             }
           }
+
+          // Transform/cast value
+          Number value = null;
+          if (rule.value != null && !rule.value.isEmpty()) {
+            String val = matcher.replaceAll(rule.value);
+
+            Matcher math = mathPattern.matcher(val);
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxCollector.java]: This is a bit more complex than I think is needed, a `valueFactor` would be sufficient I think.

Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxCollector.java]: I'd have `value` as a subsitiutable float, and `valueFactor` as number it's multiplied by. The use case is converting milliseconds to seconds, there's no need for arithmetic operations beyond that.

**Comment by icetrain** [comment]
Updated with a simpler version that only multiplies value by a factor (if supplied).

File: collector/src/main/java/io/prometheus/jmx/JmxCollector.java [hunk 2]
```diff
@@ -240,38 +248,51 @@ public void recordBean(
...
+
+            try {
+              beanValue = Double.valueOf(val);
+            } catch (NumberFormatException e) {
+              LOGGER.fine("Unable to parse configured value '" + val + "' to number for bean: " + beanName + attrName + ": " + beanValue);
+              return;
+            }
+          }
+          if (beanValue instanceof Number) {
+            if (rule.valueFactor != null) {
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxCollector.java]: Might it be simpler to default it to 1?

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**